src/modules/keycloak-administration/repository/user-lock.repository.ts

```diff
@@ -0,0 +1,69 @@
+import { EntityManager, Loaded, RequiredEntityData } from '@mikro-orm/core';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { DataConfig } from '../../../shared/config/data.config.js';
+import { ServerConfig } from '../../../shared/config/server.config.js';
+import { UserLock } from '../domain/user.lock.js';
+import { UserLockEntity } from '../entity/user-lock.entity.js';
+import { DomainError } from '../../../shared/error/domain.error.js';
+
+export function mapEntityToAggregate(entity: UserLockEntity): UserLock<true> {
+    return UserLock.construct(entity.person.id, entity.locked_by, entity.locked_until);
+}
+
+export function mapAggregateToData(userLock: UserLock<boolean>): RequiredEntityData<UserLockEntity> {
+    return {
+        person: userLock.person!,
+        locked_by: userLock.locked_by,
+        locked_until: userLock.locked_until!,
+    };
+}
+
+export function mapEntityToAggregateInplace(entity: UserLockEntity, userLock: UserLock<boolean>): UserLock<true> {
+    userLock.person = entity.person.id;
+    userLock.locked_by = entity.locked_by;
+    userLock.locked_until = entity.locked_until;
+
+    return userLock;
+}
+
+@Injectable()
+export class UserLockRepository {
+    public readonly ROOT_ORGANISATION_ID: string;
+
+    public constructor(
+        private readonly em: EntityManager,
+        config: ConfigService<ServerConfig>,
+    ) {
+        this.ROOT_ORGANISATION_ID = config.getOrThrow<DataConfig>('DATA').ROOT_ORGANISATION_ID;
+    }
+
+    public async findById(id: string): Promise<Option<UserLock<true>>> {
+        const user: Option<UserLockEntity> = await this.em.findOne(UserLockEntity, { person: id });
+        if (user) {
+            return mapEntityToAggregate(user);
+        }
+        return null;
+    }
+
+    public async createUserLock(userLock: UserLock<true>): Promise<UserLock<true> | DomainError> {
+        const userLockEntity: UserLockEntity = this.em.create(UserLockEntity, mapAggregateToData(userLock));
+        await this.em.persistAndFlush(userLockEntity);
+
+        return mapEntityToAggregateInplace(userLockEntity, userLock);
+    }
+
+    public async update(userLock: UserLock<true>): Promise<UserLock<true> | DomainError> {
+        const userLockEntity: Loaded<UserLockEntity> = await this.em.findOneOrFail(UserLockEntity, {
+            person: userLock.person,
+        });
+        userLockEntity.assign(mapAggregateToData(userLock));
+        await this.em.persistAndFlush(userLockEntity);
+        return mapEntityToAggregate(userLockEntity);
+    }
+
+    public async deleteUserLock(personId: string): Promise<Result<void, DomainError>> {
+        await this.em.nativeDelete(UserLockEntity, { person: personId });
+        return { ok: true, value: undefined };
+    }
```

Comment:
This can never produce an `ok: false`-result. If you want to transform any DB-errors into DomainErrors you'd need to catch the error thrown by the `nativeDelete`

Reply:
removed